### ISSUE TITLE
🐛 Core Web: Null error during initialization in FirebaseCoreWeb._injectSrcScript

### ISSUE THREAD

**Issue by jacobsimionato** [firebase/flutterfire#11243]
## Bug report

Google apps are seeing a crash reported on this line:
https://github.com/firebase/flutterfire/blob/6f4d3525e8521b8eea009707db50e6812c33db3a/packages/firebase_core/firebase_core_web/lib/src/firebase_core_web.dart#L142

Part of the stack trace (see internal bug for more details):
```
NullError
JsObject.callMethod (dart_lang/v2/sdk/lib/_internal/js_runtime/lib/js_patch.dart:222)
FirebaseCoreWeb._injectSrcScript (firebase_core/web/lib/src/firebase_core_web.dart:115)
_wrapJsFunctionForAsync (dart_lang/v2/sdk/lib/_internal/js_runtime/lib/async_patch.dart:307)
```
Note that the version used internally is not up to date with Github head so the line numbers above won't match, but I don't think the crashing lines have changed recently.

The crash is high frequency and affects at least 3 separate apps. Sadly, we are not able to reproduce the crash at this stage. See Google internal bug b/287826205 for more details.

We'd love any help trying to dig into this bug or guess what circumstances could trigger it.

**Comment by darshankawar** [comment]
Thanks for the report @jacobsimionato 
Seems like it is throwing the error during initialization. Can you provide us more details as what `firebase_core_web` version you are using that is throwing this error ? If you are not using the latest or atleast `2.0.0` version, is it possible to update to latest version (2.6.0) and watch again if the same error is thrown still ?

**Comment by jacobsimionato** [comment]
We're using a version that was imported into the internal repo on 14 Dec 2022, git revision c99a842f3e3f5f10246e73f51530cc58c42b49a3, which I think should be >= 2.0.0 seeing as that landed in October 2022 - https://github.com/firebase/flutterfire/blob/master/CHANGELOG.md#2022-10-19

We would like to update Firebase plugins again soon, but it's non-trivial - there seem to be some internal build infra wrangling required to get it to build.

**Comment by darshankawar** [comment]
Thanks for the feedback. Keeping this issue open based on the report and error receieved.

/cc @Lyokone

**Comment by Lyokone** [comment]
Hello @jacobsimionato,
Could it be related to the introduction of TrustedTypes that prevent the script from being injected? Support for TrustedTypes was added recently, and you might need to update to at least >= 2.2.1

**Comment by jacobsimionato** [comment]
I'm pretty sure it't can't be blocked by trusted types enforcement because this app uses "report only" mode for trusted types - I just posted the details on internal bug b/287826205.

We do really appreciate the trusted types change though, and it's on my radar to import the new version of Firebase and then enable trusted types for apps hopefully.

Any other ideas of what could cause this?

**Comment by Lyokone** [comment]
@jacobsimionato Just to make sure that you are aware, I have no access to the internal bug report of Google so I'm not able to access that.
Without a reproduction example, I'm not sure that I'll be able to help you on this

**Comment by jacobsimionato** [comment]
I apologise for continually posting links which are not useful to you! Sorry about that!

Here is all of the Firebase stack trace:
```
Translated error message:
   NullError
Deobfuscated stack trace:
  [Mapped Dart].JsObject.callMethod(dart_lang/v2/sdk/lib/_internal/js_runtime/lib/js_patch.dart:222)
  [Mapped Dart].FirebaseCoreWeb._injectSrcScript(firebase_core/web/lib/src/firebase_core_web.dart:115)
  [Mapped Dart]._wrapJsFunctionForAsync(dart_lang/v2/sdk/lib/_internal/js_runtime/lib/async_patch.dart:307)
  [Mapped Dart]._wrapJsFunctionForAsync.<anonymous>(dart_lang/v2/sdk/lib/_internal/js_runtime/lib/async_patch.dart:332)
  [Mapped Dart]._asyncStartSync(dart_lang/v2/sdk/lib/_internal/js_runtime/lib/async_patch.dart:237)
  [Mapped Dart].FirebaseCoreWeb._injectSrcScript(firebase_core/web/lib/src/firebase_core_web.dart:102)
  [Mapped Dart].FirebaseCoreWeb._injectSrcScript(firebase_core/web/lib/src/firebase_core_web.dart:102)
  [Mapped Dart].FirebaseCoreWeb._initializeCore.<anonymous>(firebase_core/web/lib/src/firebase_core_web.dart:144)
  [Mapped Dart].MappedIterator.moveNext(dart_lang/v2/sdk/lib/internal/iterable.dart:393)
  [Mapped Dart].Future.wait(dart_lang/v2/sdk/lib/async/future.dart:523)
  [Mapped Dart].FirebaseCoreWeb._initializeCore(firebase_core/web/lib/src/firebase_core_web.dart:138)
  [Mapped Dart]._wrapJsFunctionForAsync(dart_lang/v2/sdk/lib/_internal/js_runtime/lib/async_patch.dart:307)
  [Mapped Dart]._wrapJsFunctionForAsync.<anonymous>(dart_lang/v2/sdk/lib/_internal/js_runtime/lib/async_patch.dart:332)
  [Mapped Dart]._asyncStartSync(dart_lang/v2/sdk/lib/_internal/js_runtime/lib/async_patch.dart:237)
  [Mapped Dart].FirebaseCoreWeb._initializeCore(firebase_core/web/lib/src/firebase_core_web.dart:131)
  [Mapped Dart].FirebaseCoreWeb.initializeApp(firebase_core/web/lib/src/firebase_core_web.dart:170)
  [Mapped Dart]._wrapJsFunctionForAsync(dart_lang/v2/sdk/lib/_internal/js_runtime/lib/async_patch.dart:307)
  [Mapped Dart]._wrapJsFunctionForAsync.<anonymous>(dart_lang/v2/sdk/lib/_internal/js_runtime/lib/async_patch.dart:332)
  [Mapped Dart]._asyncStartSync(dart_lang/v2/sdk/lib/_internal/js_runtime/lib/async_patch.dart:237)
  [Mapped Dart].FirebaseCoreWeb.initializeApp(firebase_core/web/lib/src/firebase_core_web.dart:169)
  [Mapped Dart].FirebaseCoreWeb.initializeApp(firebase_core/web/lib/src/firebase_core_web.dart:169)
  [Mapped Dart].Firebase.initializeApp(dart/firebase_core/lib/src/firebase.dart:43)
  [Mapped Dart]._wrapJsFunctionForAsync(dart_lang/v2/sdk/lib/_internal/js_runtime/lib/async_patch.dart:307)
  [Mapped Dart]._wrapJsFunctionForAsync.<anonymous>(dart_lang/v2/sdk/lib/_internal/js_runtime/lib/async_patch.dart:332)
  [Mapped Dart]._asyncStartSync(dart_lang/v2/sdk/lib/_internal/js_runtime/lib/async_patch.dart:237)
  [Mapped Dart].Firebase.initializeApp(dart/firebase_core/lib/src/firebase.dart:43)
```

I'm sorry that we don't have a repro! We're seeing ~1500 instances of this crash per day in prod though, so we'd like to try to figure it out. The fact that it's a null error occurring deep within Firebase makes me optimistic. Do you have any idea what could be null in this fragment:

https://github.com/firebase/flutterfire/blob/6f4d3525e8521b8eea009707db50e6812c33db3a/packages/firebase_core/firebase_core_web/lib/src/firebase_core_web.dart#L142-L148

**Comment by Lyokone** [comment]
Not sure what could be null in this fragment to be honest. I'm seeing
```
    assert(document.head != null);
    document.head!.append(script);
```
just before that could cause a NullError if the head of the document does not exist, but I'm not sure in which circumstances it could happen. I'll ask to see if someone else has an idea.

**Comment by xster** [comment]
From the production stats, it seems to be primary occurring on Safari on older versions of iOS and macOS

**Comment by Lyokone** [comment]
Do you have the precise version? Maybe we can add some workaround for these versions. Might not be worth it if they are too old.

**Comment by Lyokone** [comment]
I'll close this one since we haven't had any response and went through a lot of update on Firebase Core Web since this issue. Feel free to reopen or open a new issue if you still encounter this.
